lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -0,0 +1,189 @@
+        * (https://html.spec.whatwg.org/multipage/interaction.html#blocked-by-a-modal-dialog)
+        * meaning that only the modal and its contents are interactable/focusable.
+        *
+        * Note: not supported in safari, which is why the conditional is here. Once safari gets
+        * it together, we should be able to take advantage of the "top layer" functionality built into
+        * the browser around modals to simplify the dialog styling around z-index handling
+        */
+
+        // https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029#issuecomment-968299542
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (el as any).showModal();
+      }
+      else {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (el as any).show();
+      }
+    }
+    else {
+      // without native support we don't trap focus in the dialog, but we can at least start it off there
+      el.focus();
+    }
+
+    return () => {
+      if (previouslyFocusedEl && previouslyFocusedEl instanceof HTMLElement) {
+        // The useEffect cleanup executes while the dialog is still present (in React 16 at least). While the dialog
+        // still exists, the document is still "blocked by the modal dialog" so trying to focus elements outside of
+        // it won't work. So we have to wait until the next cycle of the event loop when it's gone
+        Promise.resolve().then(() => previouslyFocusedEl.focus());
+      }
+    };
+  }, []);
+
+  // listen to the native HTMLDialogElement cancel event which supporting browsers fire when the dialog is closed
+  // via ESC
+  useEffect(function() {
+    const dialog = dialogRef.current;
+
+    if (dialog && hasNativeModalSupport) {
+      /* eslint-disable @typescript-eslint/no-non-null-assertion */
+      dialog.addEventListener('cancel', onCancel!);
+      return () => { dialog.removeEventListener('cancel', onCancel!); };
+      /* eslint-enable @typescript-eslint/no-non-null-assertion */
+    }
+    else {
+      return undefined;
+    }
+  }, [onCancel]);
+
+  const dialogContextValue = {
+    dialogEl: dialogRefState
+  };
+
+  return (
+    // Provide the dialog element to descendants so that tooltips can attach to it instead of the body,
+    // which is necessary so that they end up in the top layer rather than behind the modal
+    <DialogContext.Provider value={dialogContextValue}>
+      {
+        /* Note: role="dialog" should be redundant but I think some screenreaders (ChromeVox) don't know
+        * what a <dialog> is.  It makes a difference there.
+        *
+        * The tabindex is for the sake of browsers which don't support dialog. In those browsers we try to
+        * focus the dialog element itself when it opens which can't be done if it doesn't have a tab index
+        */
+      }
+      <dialog ref={mergedRef}
+              role={role || 'dialog'}
+              aria-modal="true"
```

Comment:
Should be based on `isModal`

Reply:
> Should be based on `isModal`

Fixed: efdce7a30f243cd7e6b5b4b659882038ff1e39bf